api/client.go

```diff
@@ -0,0 +1,102 @@
+package api
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strings"
+)
+
+// Client holds fields to make requests to a Sia API.
+type Client struct {
+	address  string
+	password string
+}
+
+// NewClient creates a new api.Client using the provided address and password.
+// If password is not the empty string, HTTP basic authentication will be used
+// to communicate with the API.
+func NewClient(address string, password string) *Client {
+	return &Client{
+		address:  address,
+		password: password,
+	}
+}
+
+// Get requests the resource at `resource` and decodes it into `obj`, returning an
+// error if requesting or decoding the resource fails.  A non-2xx status code
+// constitutes a request failure.
+func (c *Client) Get(resource string, obj interface{}) error {
+	url := "http://" + c.address + resource
```

Comment:
why, because of the `// -> /` reduction?

it's a non-issue though, I think callers know to begin routes with a leading slash.